sectionproperties/analysis/cross_section.py

```diff
@@ -1049,41 +1118,53 @@ def plot_mesh(self, ax=None, pause=True, alpha=1, materials=False, mask=None):
 
         # plot the mesh
         ax.triplot(
-            self.mesh_nodes[:, 0], self.mesh_nodes[:, 1], self.mesh_elements[:, 0:3], lw=0.5,
-            color='black', alpha=alpha, mask=mask
+            self.mesh_nodes[:, 0],
+            self.mesh_nodes[:, 1],
+            self.mesh_elements[:, 0:3],
+            lw=0.5,
+            color="black",
+            alpha=alpha,
+            mask=mask,
         )
 
         # if the material colours are to be displayed
         if materials and self.materials is not None:
             color_array = []
             legend_list = []
+            c = []  # Indices of elements for mapping colors
 
             # create an array of finite element colours
-            for element in self.elements:
+            for idx, element in enumerate(self.elements):
                 color_array.append(element.material.color)
+                c.append(idx)
 
             # create a list of unique material legend entries
             for (i, material) in enumerate(self.materials):
                 # if the material has not be entered yet
                 if i == 0 or material not in self.materials[0:i]:
                     # add the material colour and name to the legend list
-                    legend_list.append(mpatches.Patch(color=material.color, label=material.name))
+                    legend_list.append(
+                        mpatches.Patch(color=material.color, label=material.name)
+                    )
 
             cmap = ListedColormap(color_array)  # custom colormap
-            c = np.arange(len(color_array))  # indices of elements
+            # c = np.arange(len(color_array))  # indices of elements
```

Comment:
Should this be deleted, because you're populating variable c above in a different manner with the changes?